xoa/data_source_host.go

```diff
@@ -0,0 +1,36 @@
+package xoa
+
+import (
+	"github.com/ddelnano/terraform-provider-xenorchestra/client"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func dataSourceXoaHost() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceHostRead,
+		Schema: map[string]*schema.Schema{
+			"name_label": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+		},
+	}
+}
+
+func dataSourceHostRead(d *schema.ResourceData, m interface{}) error {
+	c := m.(*client.Client)
+	nameLabel := d.Get("name_label").(string)
+	hosts, err := c.GetHostByName(nameLabel)
+	if err != nil {
+		return err
+	}
+
+	if _, ok := err.(client.NotFound); ok {
+		d.SetId("")
+		return nil
+	}
```

Comment:
These conditions need to be flip flopped. `d.SetId()` will never be called because the code would return from the first block.

Reply:
you're absolutely right. what a mess. fixed.